---
title: Brent Pedersen
description: Genomics researcher
---

I'm a genomics researcher and software developer focused on creating useful, simple, fast and reliable tools for variant analysis and genomic data processing. I use rust/python/cython/nim where suitable.

**Current Focus**: Developing tools for rapid variant annotation, sample quality control, and genomic data analysis.

---

## Key Projects

<div class="grid cards" markdown>

- :material-speedometer:{ .lg .middle } **[mosdepth](https://github.com/brentp/mosdepth){:target="\_blank"}** ![GitHub stars](https://img.shields.io/github/stars/brentp/mosdepth?style=social) ![GitHub forks](https://img.shields.io/github/forks/brentp/mosdepth?style=social)
  <br> Fast BAM/CRAM depth calculation for WGS, exome, or targeted sequencing. Written in Nim for maximum performance.
  <br><hr> :material-book-open-page-variant: [Mosdepth: quick coverage calculation for genomes and exomes.](https://doi.org/10.1093/bioinformatics/btx699){:target="\_blank"}

- :material-dna:{ .lg .middle } **[somalier](https://github.com/brentp/somalier){:target="\_blank"}** ![GitHub stars](https://img.shields.io/github/stars/brentp/somalier?style=social) ![GitHub forks](https://img.shields.io/github/forks/brentp/somalier?style=social)
  <br> Fast sample-swap and relatedness checks on BAMs/CRAMs/VCFs/GVCFs. Essential for quality control in genomic studies.
  <br><hr> :material-book-open-page-variant: [Somalier: rapid relatedness estimation for cancer and germline studies using efficient sketches.](https://doi.org/10.1186/s13073-020-00761-2){:target="\_blank"}

- :material-filter:{ .lg .middle } **[slivar](https://github.com/brentp/slivar){:target="\_blank"}** ![GitHub stars](https://img.shields.io/github/stars/brentp/slivar?style=social) ![GitHub forks](https://img.shields.io/github/forks/brentp/slivar?style=social)
  <br> Genetic variant expressions, annotation, and filtering for great good. Streamlines variant analysis workflows.
  <br><hr> :material-book-open-page-variant: [Effective variant filtering and expected candidate variant yield in studies of rare human disease.](https://doi.org/10.1038/s41525-021-00227-3){:target="\_blank"}

- :material-language-go:{ .lg .middle } **[vcfanno](https://github.com/brentp/vcfanno){:target="\_blank"}** ![GitHub stars](https://img.shields.io/github/stars/brentp/vcfanno?style=social) ![GitHub forks](https://img.shields.io/github/forks/brentp/vcfanno?style=social)
  <br> Annotate a VCF with other VCFs/BEDs/tabixed files. Written in Go for speed and reliability.
  <br><hr> :material-book-open-page-variant: [VCFanno: fast, flexible annotation of genetic variants.](https://doi.org/10.1186/s13059-016-0973-5){:target="\_blank"}

- :material-language-python:{ .lg .middle } **[cyvcf2](https://github.com/brentp/cyvcf2){:target="\_blank"}** ![GitHub stars](https://img.shields.io/github/stars/brentp/cyvcf2?style=social) ![GitHub forks](https://img.shields.io/github/forks/brentp/cyvcf2?style=social)
  <br> Cython + htslib for fast VCF and BCF processing. Core library for many Python-based genomic tools.
  <br><hr> :material-book-open-page-variant: [cyvcf2: fast, flexible variant analysis with Python/Cython.](https://doi.org/10.1093/bioinformatics/btx057){:target="\_blank"}

- :material-cog:{ .lg .middle } **[echtvar](https://github.com/brentp/echtvar){:target="\_blank"}** ![GitHub stars](https://img.shields.io/github/stars/brentp/echtvar?style=social) ![GitHub forks](https://img.shields.io/github/forks/brentp/echtvar?style=social)
  <br> Using all the bits for echt rapid variant annotation and filtering. Written in Rust for maximum performance.
  <br><hr> :material-book-open-page-variant: [Echtvar: compressed variant representation for rapid annotation and filtering of SNPs and indels.](https://doi.org/10.1093/nar/gkac931){:target="\_blank"}

</div>

---

## Technical Expertise

<div class="grid cards" markdown>

- :material-language-rust:{ .lg .middle } **Rust**
  <br> Systems programming for maximum performance

- :material-language-python:{ .lg .middle } **Python/Cython**
  <br>High-performance Python with C extensions

- <img src="https://nim-lang.org/assets/img/logo.svg" alt="Nim logo" style="height:2rem;vertical-align:middle;margin-right:0.5rem;"> **Nim**
  <br> Elegant, compiled language for high-performance bioinformatics tools

- :material-dna:{ .lg .middle } **Genomics**
  <br> Variant analysis, quality control, and annotation

- :material-speedometer:{ .lg .middle } **Performance**
  <br> Optimizing algorithms and pipelines for large-scale genomic data

</div>

---

## Recent Activity

Check out the [blog section](blog/index.md) for the latest posts and updates on my research and software development work.

---

## Connect

- **GitHub**: [github.com/brentp](https://github.com/brentp)
- **Google Scholar**: [Profile](https://scholar.google.com/citations?user=tDf0VkUAAAAJ&hl=en)
- **Location**: Oregon, USA
- **Twitter**: [@brent_p](https://twitter.com/brent_p)
